### PR TITLE
Ignore inclusive descendants of `aria-hidden=true` when computing accessible names

### PR DESCRIPTION
**Pull request by Jym77**

### CODE REVIEW

File: packages/alfa-aria/test/name.spec.tsx [hunk 1]
```diff
@@ -961,3 +961,35 @@ test(`.from() correctly handles chained aria-labelledby references`, (t) => {
...
+  t.deepEqual(Name.from(heading, device).toArray(), [
+    [
+      Option.of(
+        Name.of("Hello", [
+          Name.Source.descendant(
+            heading,
+            Name.of("Hello", [
+              Name.Source.descendant(
+                exposed,
+                Name.of("Hello ", [Name.Source.data(text)])
```
Comment (Jym77) [inline packages/alfa-aria/test/name.spec.tsx:986]: Interestingly, https://w3c.github.io/accname/#step2G does not require trimming of flattening the string, which means that accessible name of text nodes is not a flat string, contrary to what https://w3c.github.io/accname/#mapping_additional_nd_te claims:
> The purpose of the computation is to create a perceivable label or description for alternative presentations, in the form of a flat string of space separated textual tokens.

🤷

Comment (Jym77) [reply packages/alfa-aria/test/name.spec.tsx:986]: https://github.com/w3c/accname/issues/63 🤣

File: packages/alfa-aria/src/name.ts [hunk 2]
```diff
@@ -521,9 +521,15 @@ export namespace Name {
...
-      // Step 2A: Is the element hidden and not referenced?
+      // Step 2A: Is the element not referenced and either hidden or not exposed?
       // https://w3c.github.io/accname/#step2A
-      if (!state.isReferencing && !isRendered(element, device)) {
+      if (
+        !state.isReferencing &&
+        (!isRendered(element, device) ||
+          element
+            .attribute("aria-hidden")
+            .some((attribute) => attribute.value === "true"))
```
Comment (kasperisager) [inline packages/alfa-aria/src/name.ts:531]: Let's stuff this in a predicate named `isHidden()` and reference https://w3c.github.io/accname/#dfn-hidden.

**Review by kasperisager — CHANGES_REQUESTED** [review]

File: packages/alfa-aria/src/name.ts [hunk 3]
```diff
@@ -521,15 +521,9 @@ export namespace Name {
...
+      // Step 2A: Is the element hidden and not referenced?
       // https://w3c.github.io/accname/#step2A
-      if (
-        !state.isReferencing &&
-        (!isRendered(element, device) ||
-          element
-            .attribute("aria-hidden")
-            .some((attribute) => attribute.value === "true"))
-      ) {
+      if (isHidden(element, device) && !state.isReferencing) {
```
Comment (kasperisager) [inline packages/alfa-aria/src/name.ts:526]:
```suggestion
      if (!state.isReferencing && isHidden(element, device)) {
```

**Review by kasperisager — CHANGES_REQUESTED** [review]

File: packages/alfa-aria/src/name.ts (hunk 3 above)
Comment (Jym77) [reply packages/alfa-aria/src/name.ts]: 🤦 🙈

**Review by kasperisager — APPROVED** [review]
Looks fantastic! 👌

**Pull request merged**